### PR TITLE
Fix use-after-free in DestroyHoisting

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci benchmark

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci benchmark

**Comment by meg-gupta** [comment]
I found a real bug due to unhandled cases with `switch_enum_addr`. Other instructions I handle in this PR like `witness_method` etc are dead and will be cleaned up by DCE later on. But the fix is needed to have a valid SIL after this pass, unless we decide to clean up such instructions in this pass.

**Review by eeckstein — CHANGES_REQUESTED** [review]
Thanks for fixing this.

Basically LGTM, I just have one change request.

File: lib/SILOptimizer/Transforms/DestroyHoisting.cpp [hunk 1]
```diff
@@ -351,6 +356,16 @@ void DestroyHoisting::getUsedLocationsOfInst(Bits &bits, SILInstruction *I) {
       // ... or abort_apply.
       getUsedLocationsOfOperands(bits, cast<AbortApplyInst>(I)->getBeginApply());
       break;
+    case SILInstructionKind::EndAccessInst:
+      getUsedLocationsOfOperands(bits,
+                                 cast<EndAccessInst>(I)->getBeginAccess());
+      break;
+    case SILInstructionKind::StructElementAddrInst:
```
[the remark this answers is not in the document]
Comment (meg-gupta) [reply lib/SILOptimizer/Transforms/DestroyHoisting.cpp:364]: Thanks, that makes sense

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci test

**Review by eeckstein — APPROVED** [review]
thanks, LGTM!

**Pull request merged**